src/service/processors.py

```diff
@@ -0,0 +1,51 @@
+from pathlib import Path
+from typing import List
+
+from xxhash import xxh32_intdigest
+
+
+def _get_partition_idx(word: str, num_partitions: int) -> int:
+    return xxh32_intdigest(word) % num_partitions
+
+
+def process_map_task(input_path: Path, num_partitions: int, output_dir: Path) -> List[Path]:
+    """
+    Processes the map task. Since it's output is split into num_partitions files, it
+    returns the output paths in a list of paths (in ascending order with regard to the partition num).
+
+    Each of the num_partitions output files will have n lines of form
+
+    "word 1"
+
+    where i-th file will contain words such that xxh32_intdigest(word) % num_partitions == i
+    and n is the number of words
+
+    For now, it represents just a simple, predetermined word counting functionality.
+
+    Simplifying assumption
+
+    * a single line of the file can fit in memory
+    * output dir must be empty at the start of this function
```

Comment:
Those will need to be dropped sometime in the future if we expect mapper to crash mid-job

Reply:
* the single line requirement can be easily solved by reading the file in chunks, but for readability I decided it's low priority and better to have something **very** simple now. In addition to that -- would we ever really get large files at this stage? For now we assume that user provides us with multiple files that map 1-1 to map tasks, but in general we'd probably split them into smaller chunks anyway
* if mapper crashes mid job, can't master assign a new directory/clean this one? We return a list of paths, so do we care? Unless you mean picking up where we left.

Reply:
I highlighted the one line thing by mistake. I think that yes, master could assign (and probably should assign) a new directory. We probably have to keep in mind the fact that we have to clean up the old one sometime